lsp-ui-doc.el

```diff
@@ -675,10 +675,14 @@ HEIGHT is the documentation number of lines."
           (setq lsp-ui-doc--timer
                 (run-with-idle-timer
                  lsp-ui-doc-delay nil
-                 (lambda nil
-                   (lsp--send-request-async
-                    (lsp--make-request "textDocument/hover" (lsp--text-document-position-params))
-                    (lambda (hover) (lsp-ui-doc--callback hover bounds (current-buffer))))))))
+                 (let ((buf (current-buffer)))
+                   (lambda nil
+                     (with-current-buffer buf
```

Comment:
I would propose checking if the new (current-buffer) is the same as buf. If it has changed - discard the change.

Reply:
Yes, the same as in https://github.com/emacs-lsp/lsp-ui/blob/e8200e3b72ecb203a854224eaf73a2194cb4ba46/lsp-ui-sideline.el#L518

Reply:
Yep, that's clearly the right thing to do. Thanks for pointing it out!